gc/scheduler/scheduler.go

```diff
@@ -255,7 +256,6 @@ func (s *gcScheduler) run(ctx context.Context) {
 		interval    = time.Second
 		gcTime      time.Duration
 		collections int
-		// TODO(dmcg): expose collection stats as metrics
```

Comment:
The number of collections is not an interesting metric on its own. The metrics should be defined in a way in which they can be used for tuning the garbage collection parameters. Frequency and average time are likely the most relevant. Delay would also be useful to track the time between a mutation/deletion and GC.

Reply:
Does Prometheus require containerd to calculate average? I generally prefer to record raw numbers and calculate summary statistics such as p50, p90 and p99 on the monitoring system side.

Reply:
Yeah, histograms can be used for that and more useful for tuning. If the goal is just monitoring or accounting, then sometimes the raw sums or averages can be easier to define. I do think it requires a little bit more thought, hence why the comment was added rather than just adding a few basic counters.

Reply:
> I generally prefer to record raw numbers and calculate summary statistics such as p50, p90 and p99 on the monitoring system side.

Agree with this. 
After search a little bit, maybe this is what needs to be done 
https://github.com/containerd/containerd/issues/1635

> - Add metric for total micros of garbage collection
> - Add metric for number of garbage collections run